docs/dev/plugins.md

```diff
@@ -148,14 +175,95 @@ If things go right, the Status now has a bunch of Activators, each with a block
 
 If things go wrong, have a look at `status.load_exception`.
 
-### Activation
+### Activation and Execution
+
+Depending on the plugin type, activation may be triggered by a number of different events. For example, CliCommand plugin types are activated when their activation name is mentioned in the command line arguments.
+
+After activation, code for that aspect of the plugin is loaded and ready to execute. Executioon may be triggered by a number of different events. For example, the CliCommand plugin types are implicitly executed by Thor when `Inspec::CLI` calls `start()`.
+
+Refer to the sections below for details about activation and execution timing.
+
+## Implementing a CLI Command Plugin
+
+The CliCommand plugin_type allows you to extend the InSpec command line interface by adding a namespace of new commands. InSpec is based on [Thor](http://whatisthor.com/) ([docs](https://www.rubydoc.info/github/wycats/thor/Thor)), and the plugin system exposes Thor directly.
+
+CliCommand can do things like:
+
+```bash
+# A namespaced custom command with options
+you@machine$ inspec sweeten add --kind sugar --teaspoons 2
+# A namespaced custom command with short options
+you@machine$ inspec sweeten add -k agave
+# Mix global and namespace options
+you@machine$ inspec --debug sweeten add -k aspartame
+# Namespace included in help
+you@machine$ inspec help
+Commands:
+  inspec archive PATH      # archive a profile to tar.gz (default) or zip
+  inspec sweeten ...       # Add spoonfuls til the medicine goes down
+# Detailed help
+[cwolfe@lodi inspec-plugins]$ be inspec help sweeten
+Commands:
+  inspec sweeten add [opts]       # Adds sweetener to your beverage
+  inspec sweeten count            # Reports on teaspoons in your beverage, always bad news
+```
+
+Currently, it cannot do a direct (non-namespaced) command
+```bash
```

Comment:
I think we can remove this and following space.

Reply:
Thanks, that was leftover from a merge conflict manual resolution - good catch!